States/graph.c

```diff
@@ -0,0 +1,203 @@
+#include <stdlib.h>
+#include "graph.h"
+#include <stdio.h>
+#include <math.h>
+#include <stdbool.h>
+#include <string.h>
+
+void setValueTo(int* array, int length, int value) {
+    for (int i = 0; i < length; ++i) {
+        array[i] = value;
+    }
+}
+
+bool allIsVisited(bool* visitedCities, bool* allowedCities, int citiesCount) {
+    for (int i = 0; i < citiesCount; i++) {
+        if (visitedCities[i] == false && allowedCities[i] == true) {
```

Comment:
```suggestion
        if (!visitedCities[i] && allowedCities[i]) {
```
Стайлгайд запрещает сравниваться с булевыми константами. Так и в production, бывает, пишут, но это демонстрирует неуважение к булевой алгебре, на матмехе такое не любят.